tasktiger/redis_scripts.py

```diff
@@ -330,11 +346,24 @@ def can_replicate_commands(self) -> bool:
             self._can_replicate_commands = result
         return self._can_replicate_commands
 
-    def register_script_from_file(self, filename: str) -> Script:
+    def register_script_from_file(
+        self, filename: str, include_functions: Optional[dict] = None
+    ) -> Script:
         with open(
             os.path.join(os.path.dirname(os.path.realpath(__file__)), filename)
         ) as f:
-            return self.redis.register_script(f.read())
+            script = f.read()
+            if include_functions:
+                function_definitions = []
+                for func_name, func_body in include_functions.items():
```

Comment:
Since this is a dict, the order isn't technically guaranteed (although I believe the order is consistent in later Python versions). Just to be safe we should sort the keys to avoid storing multiple similar script with different function orders in Redis.

Reply:
great point, will fix